redis/src/client.rs

```diff
@@ -287,6 +290,91 @@ impl Client {
+    ///
+    /// If client cert+key pair is not provided, then client-side authentication is not enabled. If
+    /// trust anchors are not provided, then system root certificates are used instead.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::{fs::File, io::{BufReader, Read}};
+    ///
+    /// use redis::{Client, AsyncCommands as _, CertificatesBinary, ClientTlsBinary};
+    ///
+    /// async fn do_redis_code(
+    ///     url: &str,
+    ///     root_cert_file: &str,
+    ///     cert_file: &str,
+    ///     key_file: &str
+    /// ) -> redis::RedisResult<()> {
+    ///     let root_cert_file = File::open(root_cert_file).expect("cannot open private cert file");
+    ///     let mut root_cert_vec = Vec::new();
+    ///     BufReader::new(root_cert_file)
+    ///         .read_to_end(&mut root_cert_vec)
+    ///         .expect("Unable to read ROOT cert file");
+    ///
+    ///     let cert_file = File::open(cert_file).expect("cannot open private cert file");
+    ///     let mut client_cert_vec = Vec::new();
+    ///     BufReader::new(cert_file)
+    ///         .read_to_end(&mut client_cert_vec)
+    ///         .expect("Unable to read client cert file");
+    ///
+    ///     let key_file = File::open(key_file).expect("cannot open private key file");
+    ///     let mut client_key_vec = Vec::new();
+    ///     BufReader::new(key_file)
+    ///         .read_to_end(&mut client_key_vec)
+    ///         .expect("Unable to read client key file");
+    ///
+    ///     let client = Client::build_with_tls(
+    ///         url,
+    ///         CertificatesBinary {
+    ///             client_tls: Some(ClientTlsBinary{
+    ///                 client_cert: client_cert_vec,
+    ///                 client_key: client_key_vec,
+    ///             }),
+    ///             root_cert: Some(root_cert_vec),
+    ///         }
+    ///     )
+    ///     .expect("Unable to build client");
+    ///
+    ///     let connection_info = client.get_connection_info();
+    ///
+    ///     println!(">>> connection info: {connection_info:?}");
+    ///
+    ///     let mut con = client.get_async_connection().await?;
+    ///
+    ///     con.set("key1", b"foo").await?;
+    ///
+    ///     redis::cmd("SET")
+    ///         .arg(&["key2", "bar"])
+    ///         .query_async(&mut con)
+    ///         .await?;
+    ///
+    ///     let result = redis::cmd("MGET")
+    ///         .arg(&["key1", "key2"])
+    ///         .query_async(&mut con)
+    ///         .await;
+    ///     assert_eq!(result, Ok(("foo".to_string(), b"bar".to_vec())));
+    ///     println!("Result from MGET: {result:?}");
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    #[cfg(feature = "tls-rustls")]
+    pub fn build_with_tls<C: IntoConnectionInfo>(
+        conn_info: C,
+        tls_certs: CertificatesBinary,
+    ) -> RedisResult<Client> {
+        inner_build_with_tls(conn_info, tls_certs)
```

Comment:
Another nitpick, but there isn't really a need for `inner_build_with_tls` to be generic. `build_with_tls` should be the one to call `into_connection_info`. This will help with compilation times and binary sizes in some situations.

You probably won't notice the difference in most situations (again, nitpick), but it's a good practice and it's easy enough in this case.